### DISCUSSION TITLE
Boost the developer rewards program with donations

### DISCUSSION THREAD

**Discussion by alaznem** [RoboSats/robosats#1761] [General]
I like the current state of the developer rewards panel very much and I think there is a potential to acquire more spendable sats by donations.

I could imagine a small communication campaign issued by @'RoboSats on twitter with the context of either

a) adding specific features (github issues as idea pool) the users are willing to boost or
b) enabling some heavy lifting by accepting donations not pegged to a specific task. Maybe categorized to a few types like "conceptual work", "implementation", "to assing freely on anything", ...

I offer to manage the inflow of the donated sats (issuing invoices to willing donators) in the pilot phase as long as a 3rd party custodial key storage is acceptable or I get access to corresponding robosats infrastructure. This is for my own privacy and convenience to not have to spin up a fresh node dedicated to this purpose and not to bother with setting up a BTCPay Server without the confirmation that donations will flow in at all.

In addition I would provide conceptual work about the interface between funds inflow and funds allocation (which sats are allocable to which purpose/issue/task/feature).

About the explicit campaign concept and final communication content I'm two-part: not burning to do it myself but if no one jumps on this I would have fun to do it, just to make donaton this test happen. 

Furthermore, if this idea gains attraction, a next step could be to propose/ implement an optional "mark one checkbox to make a donation (absolute or relative to the trade amount) if your trade is successful" within the trade workflow of robosats.

About the questions if more spendable sats attracts more (developer) resources or if more bounties make the current contributors happier, I didn't think about at all.

What I hope would happen is to encourage people which don't contribute to bitcoin at all currently, to start doing so. What I don't want to achieve is to redirect a) funds from existing bitcoin donors towards robosats and lower donations for other projects or b) take away developer resources from other bitcoin projects by being more attractive for developers on the sat rewarding side.

**Comment by Reckless-Satoshi** [comment]
Hey @alaznem these are great ideas! Also nice to see you want to get in charge of some of these tasks.

We have an under-promoted guide on how to donate to RoboSats https://learn.robosats.com/contribute/donate/ . Would you propose directly how to adapt your specific suggestions? This donation guide is underpromoted on purpose: I do not think RoboSats should actively ask for donations. However, I think it is a great idea if users want to donate to specific features to get them done faster.

---------
Rewards could be paid directly from **a) the central Dev Fund** or **b) the donnor straight to the dev**.

**a)** These are paid out directly by [RoboSats node](https://amboss.space/search?term=cm9ib3NhdHM=), so it is easiest if donnors send Keysend to RoboSats (a message can be attached). However, many users cannot do keysend, in that case it would be very **helpful if you want to take the role of collector** (make invoices, then Keysend to RoboSats or directly pay the devs). 

**b)** The donnor can directly participate on Github Issues/PR and ask the developer for an invoice. **Cons**: these rewards are hard to add to the Rewards Panel.

-------
As for withdrawing resources (donation/developers) from other bitcoin projects, I think it is important to keep in mind. However, it is too early to evaluate that impact. At the moment RoboSats operation and development is heavily subsidized by me.

**Comment by alaznem** [comment]
> We have an under-promoted guide on how to donate to RoboSats https://learn.robosats.com/contribute/donate/

This guide content makes sense to me. I didn't know about it before, thank you for pointing me there. I will let this pages content sink in and come back here when/if they synthesized with my initial thoughts about donations for robosats.

Since you don't want to promote or use the above donations guide heavily, the following thoughts might be obsolete and you probably know about this already and are willing to accept the tradeoffs for having things easy set up:

- '1. Directly tip the developer by an invoice the developer provides does public leak the developers node pubkey, if they self custody.
- '3. Looks like a good old bad invoice reuse privacy debacle.
- '4. The widget doesn't work in desktop tor browser for me, because of the error "No wallet available \n You first need to install a browser extension."

**Comment by Reckless-Satoshi** [comment]
> * '1. Directly tip the developer by an invoice the developer provides does public leak the developers node pubkey, if they self custody.

Indeed, developers must share an invoice from a proxy wallet to not leak their pubkey.

> * '4. The widget doesn't work in desktop tor browser for me, because of the error "No wallet available \n You first need to install a browser extension."

I have not tested myself with Tor Browser. I believe you need a WebLN compatible extension, e.g. [Alby](https://getalby.com/) should work on Tor Browser.
